lib/neo4j/core/query_clauses.rb

```diff
@@ -643,9 +643,7 @@ def from_key_and_value(key, value)
           case value
           when /^:/
             "#{key}:`#{value[1..-1]}`"
-          when String
-            "#{key}.#{value}"
-          when Symbol
+          when String, Symbol
```

Comment:
this change does not make sense. 
```
"#{key}.#{value}"
```
 is not the same as  
```
"#{key}:`#{value}`"
```
The former references a property the latter tests a node if it has a label. Use dot notation with escape and that probably will be good although readability of logs may suffer. Maybe there is a way to escape only when necessary.

Reply:
Oh I completely missed that there was a different delimiter there, good catch, that is probably also a semi-significant cause of the test errors. It seems like escape handling isn't really standardized, I discovered this particular issue in a chained method that ended in `.delete`, where it properly escaped everything up until then.

I'm not sure if it makes sense to check if a string _needs_ to be delimited, that is a recipe for injection attacks, and it would incur a slight performance penalty.

This is my first contribution, but it also seems strange to have a distiction between passing a string vs. a symbol to delete. The documentation for `delete` seems to indicate that there shouldn't be a difference: https://neo4jrb.readthedocs.io/en/v9.4.0/QueryClauseMethods.html#delete. If there should be a distinction, it should be documented.